include/glow/ExecutionEngine/Context.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_EXECUTIONENGINE_CONTEXT_H
+#define GLOW_EXECUTIONENGINE_CONTEXT_H
+
+#include "glow/Base/Tensor.h"
+
+#include <unordered_map>
+
+namespace glow {
+
+class Tensor;
+class Storage;
+class Placeholder;
+
+/// This class provides a mapping between some graph nodes, which are a symbolic
+/// representation of some computation, and concrete tensors that represent the
+/// inputs and outputs to the graph. The context owns the tensors and the graph
+/// uses these values as runtime. This is useful for the multi-threaded
+/// execution of code, where each thread has a different execution context. The
+/// difference between this class and a regular map is that the Context owns the
+/// Tensors (not only the pointers) and manages their lifetime.
+class Context final {
+  /// Maps Placeholders to Tensors.
+  std::unordered_map<Placeholder *, Tensor *> map_;
+
+public:
+  /// \returns the tensor that corresponds to Placeholder \p P or Null if the
+  /// tensor is not found.
+  Tensor *get(Placeholder *P);
+
+  /// Inserts the Placeholder-Tensor pair.
+  void insert(Placeholder *P, Tensor &&T);
+
+  /// \returns True if \p P is a registered Placeholder.
+  bool count(Placeholder *P);
```

Comment:
Having a function called 'count()' return a bool is a little weird.  :-)

Reply:
It seems like map::count() returns zero or one. I think that it's okay to return bool because this is a new wrapper API. But we way want to change the name?  What do you think? 


From the documents of unordered map:  
"
Because unordered_map containers do not allow for duplicate keys, this means that the function actually returns 1 if an element with that key exists in the container, and zero otherwise.
"

Reply:
Acknowledged.  I do not feel strongly about this kind of thing.  Just wanted to point it out.  :-)

Reply:
Okay. I changed this to size_t.